### ISSUE TITLE
Reduce DOM addition

### ISSUE THREAD

**Issue by Yoz0** [Sindarus/tsumegomaker#21]
According to the [JQuery guideline ](http://learn.jquery.com/performance/append-outside-loop/) it isn't recommanded to add lots of element to the DOM. Iirc this is exactly what we're doing maybe this affect the performance of the app.

**Comment by Sindarus** [comment]
In javascript, we could easily apply the techniques explained in the link provided above.
Or, I think we could find a way to create the html server side and then hand it to the client, this way we would have no DOM addition at all, this would imply building an html string in ruby I think.
When we fix this issue, maybe we could seize the opportunity to change the display system, to switch to SVG for example. Seriously, we currently have a table of images, that's a bit crappy. ^^

**Comment by Sindarus** [comment]
Or maybe we could just quick fix this with the techniques explained in the link and leave the rest of the work for later.

**Comment by Yoz0** [comment]
Creating the Html server side isn't really MVC-style... And adding a bit to the DOM is totally ok.

If we change the display system, we need to discuss what are the improvement we want. We shouldn't change the display system only to change the display system. I do think we need a more responsive system though.

**Comment by Sindarus** [comment]
Uhm, I don't understand why you say creating html server side isn't mvc-style, because I think that's what we're doing right now for every page, and it's what every MVC website is doing, it seems.
I think the request/response cycle is the following : when someone sends a get request to the server, it executes the controller, and then generates the html page using haml, and only then it sends it back.
So actually I think it'd be even more MVC-style than what we're doing right now, because the view would be created all in the same place, instead of being relocated to the client.

Still I agree that adding one single element to the DOM is fine for now.
Sure, I was thinking of improving responsiveness too, by using svg.

**Comment by Yoz0** [comment]
You're right, we create Html server side. Yet we shouldn't create an Html string in the controller, that is not MVC. We could make a view that create the table, seems a good idea to me, but not really compatible with websockets.

**Comment by Sindarus** [comment]
Sure sure, only a view should take care of that.
I don't see the problem with websockets. To me, websockets are used for communicating moves, board data (not html), legal moves and such. I don't understand how building view server side would not be websockets compatible.
